annif/cli.py

```diff
@@ -408,27 +449,21 @@ def run_index(
     backend_params = parse_backend_params(backend_param, project)
     hit_filter = SuggestionFilter(project.subjects, limit, threshold)
 
-    for docfilename, dummy_subjectfn in annif.corpus.DocumentDirectory(
-        directory, project.subjects, project.vocab_lang, require_subjects=False
-    ):
-        with open(docfilename, encoding="utf-8-sig") as docfile:
-            text = docfile.read()
+    documents = annif.corpus.DocumentDirectory(
+        directory, None, None, require_subjects=False
+    )
+    subject_sets = project.suggest_batch(documents, backend_params)
+
+    for (docfilename, dummy_subjectfn), subjects in zip(documents, subject_sets):
```

Comment:
`_` could be used here insted of `dummy_subjectfn`, as `_` is commonly used for throwaway variables